apps/gallery/js/ImageEditor.js

```diff
@@ -1221,8 +1230,51 @@ ImageEditor.prototype.getFullSizeBlob = function(type, done, progress) {
       setTimeout(function() {
         done(blob);
       });
+    }
+
+    function updateExif(blob, metaData, callback) {
+      metaData.Orientation = 1;
+      if (metaData.PixelXDimension) {
+        metaData.PixelXDimension = canvas.width;
+      }
+      if (metaData.PixelYDimension) {
+        metaData.PixelYDimension = canvas.height;
+      }
+      JPEGParser.writeExifMetaData(
+        blob, metaData,
+        (error, modifiedBlob) => {
+          if (error) {
+            console.error('Error' + error);
+          }
+          // Process modified file
+          callback(modifiedBlob);
+        });
+    }
+
+    var originalBlob = this.imageBlob;
+    var originalExif = this.edits.EXIF;
+
+    // Finally, convert the edited image to a blob and pass to the callback.
+    canvas.toBlob(function(blob) {
+      if (type == 'image/jpeg') {
+        if (originalExif) {
+          updateExif(blob, originalExif, gotEncodedBlob);
+        } else {
+          LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
```

Comment:
It might be nice to have a comment here explaining that originalExif is almost certainly already set and that this else clause is only here in case the original call did not return yet.